core/Pos/Core/Block/Union/Instances.hs

```diff
@@ -26,7 +26,9 @@ import           Pos.Core.Slotting.Types (EpochOrSlot (..))
 
 instance Bi BlockHeader =>
          Buildable BlockHeader where
-    build = either Buildable.build Buildable.build
+    build = \case
+        BlockHeaderGenesis bhg -> Buildable.build bhg
+        BlockHeaderMain    bhm -> Buildable.build bhm
```

Comment:
Maybe there's sense in having `either`-like combinator for `BlockHeader`? Though, I don't know how to name it...

Reply:
One option is `view $ choosingBlockHeader (to Buildable.build) (to Buildable.build)`, but I decided not to bother.